src/runtime/container.rs

```diff
@@ -161,3 +113,53 @@ impl Container {
         None
     }
 }
+
+impl TryFrom<Vec<ContainerElement>> for Container {
+    type Error = ContainerError;
+
+    fn try_from(mut elements: Vec<ContainerElement>) -> Result<Container, ContainerError> {
+        use ContainerElement as CE;
+        // take last element of Container
+        let data = match elements.pop() {
+            Some(CE::SpecialFinal(Some(data))) => data,
```

Comment:
this is more for my readability, but could you keep `ContainerElement` instead of CE?